langlib/lang.map/src/main/java/org/ballerinalang/langlib/map/util/MapLibUtils.java

```diff
@@ -62,6 +62,7 @@ public static BType getCommonTypeForRecordField(BRecordType recordType) {
             typeSet.add(recordType.restFieldType);
         }
 
-        return typeSet.size() == 1 ? typeSet.iterator().next() : new BUnionType(typeSet.toArray(emptyTypeArr));
+        return typeSet.size() == 1 ? typeSet.iterator().next()
+                : new BUnionType(typeSet.toArray(emptyTypeArr), recordType.typeFlags);
```

Comment:
Setting `recordType.typeFlags` could be incorrect here right?

For example
```ballerina
type Foo record {
    string s;
    error e;
};
```

Both TypeFlags.ANYDATA  and TypeFlags.PURETYPE would be set for this record, but setting the same for the union created here (`string|error`) is incorrect, since this not `anydata`?